tests/update/patch_sle.pm

```diff
@@ -173,7 +173,15 @@ sub sle_register {
     # SLE 12 and later use SCC, but SLE 11 uses NCC
     if ($action eq 'register') {
         if (is_sle('12+')) {
+            # Tag the test as being called from this module, so accept_addons_license
+            # (called by yast_scc_registration) can handle license agreements from modules
+            # that do not show license agreement during installation but do when registering
+            # after install
+            set_var('IN_PATCH_SLE', 1);
```

Comment:
still think that it should be solved by only modifying `@SLE15_ADDONS_WITHOUT_LICENSE` and without introducing new arrays/variables. Just need to find right spot where it should be done. Maybe here ?
```suggestion
        @SLE15_ADDONS_WITHOUT_LICENSE =    grep {!/ha/} @SLE15_ADDONS_WITHOUT_LICENSE;
                @SLE15_ADDONS_WITHOUT_LICENSE =    grep {!/we/} @SLE15_ADDONS_WITHOUT_LICENSE;
```

you need to ask someone else how properly remove several elements by value from perl array :) but this won't change my idea

Reply:
Yeah, it's not easy. I thought about deciding what to do depending on the value of `PATCH`, but since this code is shared between SCC registrations during `patch_sle` and during install, if I do that it will fix the registration during `patch_sle` but break the registration during upgrade (if there's any). The problem is that during installation there is no license agreement screen for HA and WE (http://mango.suse.de/tests/709#step/scc_registration/16 and http://mango.suse.de/tests/709#step/scc_registration/17), but during patch there is (http://mango.suse.de/tests/706#step/patch_sle/27 and http://mango.suse.de/tests/706#step/patch_sle/28) and the code handling both cases is the same.

Reply:
My first attempt did a `push @addons_with_licenses, qw(ha we) if (is_sle('15+') and get_var('IN_PATCH_SLE'));`, but then thought it would be more readable with a new array.